domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt

```diff
@@ -180,11 +191,29 @@ class ExplorationProgressController @Inject constructor(
               explorationProgress.stateGraph.getState(answerOutcome.stateName),
               prohibitSameStateName = true
             )
+            // Reset the hintState if pending top state has changed.
+            explorationProgress.hintState = hintHandler.reset()
+          } else {
+            //  Schedule, or show immediately, a new hint or solution based on the current ephemeral
```

Comment:
Dont we need test cases in ExplorationProgressControllerTest for this scenario?

Reply:
added more tests to test this.

Reply:
This appears to be resolved (tests in ExplorationProgressControllerTest include cases for wrong answer submission--I verified by removing the new version of this code & observed 9 test failures in ExplorationProgressControllerTest). Closing the thread with permission from @vinitamurthi to verify & close out old conversation threads.